src/internal/storage/chunk/writer.go

```diff
@@ -89,7 +89,8 @@ func newWriter(ctx context.Context, client Client, memCache kv.GetPut, deduper *
 		},
 		buf:   &bytes.Buffer{},
 		stats: &stats{},
-		chain: NewTaskChain(cancelCtx),
+		// TODO: Make task chain parallelism configurable?
+		chain: NewTaskChain(cancelCtx, 100),
```

Comment:
Don't we bottleneck on the rolling hash here?  Might make sense to set this to `runtime.GOMAXPROCS(0)` if it's CPU bound.

Reply:
The rolling hash can be the bottleneck in certain cases, but it wouldn't be in the copy case since we don't go through the rolling hash in most cases (the concat step which copies the shards and uses the rolling hash for the edges of the shards). Even in the case where we are not copying, if the upload to the object storage provider is slower than the rolling hash, we will just keep buffering chunks in memory until the full data has been written (assuming the data is coming into the writer fast enough, which is the case for the compact at the close of an unordered writer).